app/models/request_geolocation_info.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Decorates a Geocoder:Result object to consistify its interface
+# with that of `Geocodeable` ActiveRecord objects.
+# see: app/models/concerns/geocodeable.rb
+class RequestGeolocationInfo < SimpleDelegator
```

Comment:
I think there is already a class that is suppose to do this - `Geohelper` - which also is suppose to manage the differences between test responses, production responses, different Google responses, and Maxmind responses (we're also going to be adding/switching to mapbox soon, so it will need to handle all that as well)

Reply:
This intent for this class is primarily to map `country` messages to `country_code` and `zipcode` to `postal_code`, to match the other receivers, which are ActiveRecord objects. I don't see a way for `Geohelper` to do that.

Reply:
Is there a way to put it into `Geocodeable`?

Reply:
No prob — updated in 1b10da8e : 

```rb
# app/services/geohelper.rb L61-83 (1b10da8e)

# Extract location info from the given result object `result`,
# expected to be an ActiveRecord or Geocoder::Result object.
#
# Returns a Hash
def location_from_result(result)
  location = {
    address: result.address.presence,
    city: result.city.presence,
    latitude: result.latitude.presence,
    longitude: result.longitude.presence,
  }

  if result.respond_to?(:country_code)
    location.merge(
      country: Country.fuzzy_find(result.country_code),
      zipcode: result.postal_code.presence,
    )
  else
    location.merge(
      country: result.country.presence,
      zipcode: result.zipcode.presence,
    )
  end
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/1b10da8e/app/services/geohelper.rb#L61-L83)]</sup>